nodedb.go

```diff
@@ -84,7 +85,108 @@ type nodeDB struct {
 	fastNodeCacheQueue *list.List               // LRU queue of cache elements. Used for deletion.
 }
 
-func newNodeDB(db dbm.DB, cacheSize int, opts *Options) *nodeDB {
+func (ndb *nodeDB) getOpts() Options {
+	return ndb.opts
+}
+
+func (ndb *nodeDB) setOpts(options Options) {
+	ndb.opts = options
+}
+
+func (ndb *nodeDB) getBatch() dbm.Batch {
+	return ndb.batch
+}
+
+func (ndb *nodeDB) setBatch(batch dbm.Batch) {
+	ndb.batch = batch
+}
+
+func (ndb *nodeDB) getDb() dbm.DB {
+	return ndb.db
+}
+
+func (ndb *nodeDB) setDb(db dbm.DB) {
+	ndb.db = db
+}
+
+func (ndb *nodeDB) setStorageVersion(version string){
+	ndb.storageVersion = version
+}
+
+func (ndb *nodeDB) getFastNodeCache() map[string]*list.Element{
+	return ndb.fastNodeCache
+}
+
+type NodeDB interface {
+	getFastNodeCache() map[string]*list.Element
+	setStorageVersion(version string)
+	getDb() dbm.DB
+	setDb(db dbm.DB)
+	getOpts() Options
+	setOpts(options Options)
+	GetNode(hash []byte) *Node
+	GetFastNode(key []byte) (*FastNode, error)
+	SaveNode(node *Node)
+	SaveFastNode(node *FastNode) error
+	SaveFastNodeNoCache(node *FastNode) error
+	setFastStorageVersionToBatch() error
+	getStorageVersion() string
+	hasUpgradedToFastStorage() bool
+	shouldForceFastStorageUpgrade() bool
+	saveFastNodeUnlocked(node *FastNode, shouldAddToCache bool) error
+	Has(hash []byte) (bool, error)
+	SaveBranch(node *Node) []byte
+	resetBatch() error
```

Comment:
Thanks for the PR. Just a quick note. We should only be adding exported methods to the interface. If there are places where we call unexported methods of `nodedb` (such as  `immutable tree`, that should be fixed by either:
- finding an exported method to accomplish the task
- exporting the private method